### PR TITLE
(CFACT-263) Fix child process cleanup for POSIX command timeout.

### PR DESCRIPTION
**Pull request by peterhuene**
Fixing POSIX command timeout to properly kill the entire child process
tree rather than just the child.  This will kill other processes spawned
from the child process.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
:+1:

File: lib/tests/execution/posix/execution.cc [hunk 1]
```diff
@@ -378,8 +388,17 @@ SCENARIO("executing commands with execution::each_line") {
     GIVEN("a long-running command") {
         WHEN("given a timeout") {
             THEN("a timeout exception should be thrown") {
-                option_set<execution_options> options = { execution_options::defaults };
-                REQUIRE_THROWS_AS(each_line("sh", { LIBFACTER_TESTS_DIRECTORY "/fixtures/execution/sleep.sh" }, [](string& line) { return true; }, options, 1), timeout_exception);
+                try {
+                    option_set<execution_options> options = { execution_options::defaults };
+                    each_line("sh", { LIBFACTER_TESTS_DIRECTORY "/fixtures/execution/sleep.sh" }, [](string& line) { return true; }, options, 1);
```
Comment (MikaelSmith) [inline lib/tests/execution/posix/execution.cc]: The Windows equivalent will be "cmd.exe /c ruby.exe -e 'sleep 60'". To test killing a process tree, I'm going to change the Windows tests to that. See any reason we'd also want to test single-process kills?

Comment (peterhuene) [inline lib/tests/execution/posix/execution.cc]: I do not.

**Comment by peterhuene** [comment]
Looks like the check that the process group was killed wasn't as reliable as I thought.  Will rework.

**Comment by peterhuene** [comment]
Ignore coveralls failure.

File: lib/tests/execution/posix/execution.cc [hunk 2]
```diff
@@ -231,7 +233,17 @@ SCENARIO("executing commands with execution::execute") {
         WHEN("given a timeout") {
             THEN("a timeout exception should be thrown") {
                 option_set<execution_options> options = { execution_options::defaults };
-                REQUIRE_THROWS_AS(execute("sh", { LIBFACTER_TESTS_DIRECTORY "/fixtures/execution/sleep.sh" }, options, 1), timeout_exception);
+                try {
+                    execute("sh", { LIBFACTER_TESTS_DIRECTORY "/fixtures/execution/sleep.sh" }, options, 1);
+                    FAIL("did not throw timeout exception");
+                } catch (timeout_exception const& ex) {
+                    // Verify the process group was killed (killing again should error)
```
Comment (MikaelSmith) [inline lib/tests/execution/posix/execution.cc]: This comment is no longer accurate.

File: lib/tests/execution/posix/execution.cc [hunk 3]
```diff
@@ -378,8 +390,18 @@ SCENARIO("executing commands with execution::each_line") {
...
         WHEN("given a timeout") {
             THEN("a timeout exception should be thrown") {
-                option_set<execution_options> options = { execution_options::defaults };
-                REQUIRE_THROWS_AS(each_line("sh", { LIBFACTER_TESTS_DIRECTORY "/fixtures/execution/sleep.sh" }, [](string& line) { return true; }, options, 1), timeout_exception);
+                try {
+                    option_set<execution_options> options = { execution_options::defaults };
+                    each_line("sh", { LIBFACTER_TESTS_DIRECTORY "/fixtures/execution/sleep.sh" }, [](string& line) { return true; }, options, 1);
+                    FAIL("did not throw timeout exception");
+                } catch (timeout_exception const& ex) {
+                    // Verify the process group was killed (killing again should error)
```
Comment (MikaelSmith) [inline lib/tests/execution/posix/execution.cc]: Comment's no longer true.

**Comment by MikaelSmith** [comment]
Checking the group terminated isn't perfect, because it could potentially be re-used. Seems unlikely on that short a time-frame (famous last words).

**Comment by peterhuene** [comment]
Comments updated.

**Pull request merged**